services/rds/src/main/java/org/finra/gatekeeper/services/aws/RdsIamAuthService.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023. Gatekeeper Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.finra.gatekeeper.services.aws;
+
+import com.amazonaws.services.rds.AmazonRDSClient;
+import com.amazonaws.services.rds.auth.GetIamAuthTokenRequest;
+import com.amazonaws.services.rds.auth.RdsIamAuthTokenGenerator;
+import com.amazonaws.services.rds.model.*;
+import org.finra.gatekeeper.services.aws.model.AWSEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+/**
+ * Service that handles interfacing with the AWS for everything RDS IAM Authentication related
+ */
+@Component
+public class RdsIamAuthService {
+    private final Logger logger = LoggerFactory.getLogger(RdsIamAuthService.class);
+
+    private final AwsSessionService awsSessionService;
+
+    @Autowired
+    public RdsIamAuthService(AwsSessionService awsSessionService) {
+        this.awsSessionService = awsSessionService;
+    }
+
+
+    public boolean isRdsIamAuthEnabled(AWSEnvironment environment, String dbInstanceIdentifier, String instanceName, String instanceType) {
+        logger.info(dbInstanceIdentifier);
```

Comment:
Let's give this log more pertinent info, like "Checking to see if RDS IAM Auth is enabled for instance: "${instance}"